extension/dynamicconfig/service/service.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"context"
+	"errors"
+
+	pb "github.com/vmingchen/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+)
+
+// ConfigBackend defines a general backend that the service can read
+// configuration data from.
+type ConfigBackend interface {
+	GetFingerprint() []byte
+	IsSameFingerprint(fingerprint []byte) bool
```

Comment:
Go recommends small interfaces. Is this "IsSameFingerprint" needed? A byte-wise comparison of GetFingerprint() should be okay right?

Reply:
Great point; removing method